tdee_macro_calc/main.py

```diff
@@ -1,81 +1,40 @@
-    # (optional, maybe just default to .4) Per-lb fat intake (.4-.5)
-
-# Output
-    # Current TDEE (average of current daily calories)
-    # Current average weight
-    # Target daily calorie deficit (3.2kcal/lb)
-    # Target daily calorie intake (Current TDEE minus target daily calorie deficit)
-    # Macros - first hit the calorie goal, then protein, then fat, then fill rest with carbs
-        # Protein 4 cal per gram, .8-1.0g/lb for building muscle, 120g-160g/day is a good place to start
-            # Output: how many grams of protein you should eat (and as an FYI, how many calories that is)
-        # Fat 9 cal per gram, minimum .3g/lb
-            # Output: how many grams of fat you should eat (and as an FYI, how many calories that is)
-        # Carbs 4 cal per gram Essentially, take the calories eaten in protein and fat, subtract from your goal calories per day, and the rest is what you can eat in carbs. However, carbs/fat really doesn't matter, what is most important is protein.
-            # Output: how many grams of carbs you should eat (and as an FYI, how many calories that is)
-
-# JSON Data structure DRAFT
-data = {
-    "user": {
-        "current_weight": 176, # this will be updated with a rolling 7-day average
-        "current_tdee": 2400, # this will be updated with a max 4 week (?) average
-        "sex": "male", # this determines default of 3.2, if female can go to 4.5 I believe. Not sure if I want to actually include this as functionality right now or not
-        "target_calorie_intake": 1700,
-        "per_lb_calorie_deficit": 3.2, # use 3.2 as default but allow user to change, possibly. TODO look at study to confirm dropping this number works
-        "per_lb_protein": .8, # use .8 as default but allow user to change, and/or play with different numbers in memory
-        "per_lb_fat": .3 # use .3 as default but allow user to change, and/or play with different numbers in memory
-    },
-    "tdee": [ # TODO This could totally be an object, same with user
-        {
-            "date": "9/20/22", # TODO this should be a datetime object and need to handle type checking
-            "weight": 178.1,
-            "calories": 2531
-        },
-        {
-            "date": "9/21/22",
-            "weight": 177.3,
-            "calories": 1971
-        },
-        {
-            "date": "9/22/22",
-            "weight": 179.2,
-            "calories": 2391
-        }
-    ]
-}
-
-def add_to_file():
+def check_data_file() -> dict:
+    """
+    Checks to see if the file data.json exists, and if not it creates it with default values.
+
+    Returns a dictionary of the user and tdee information in the file.
+    """
     try:
         with open("data.json") as f:
-            new_data = json.load(f)
-            print(new_data)
-    except:
-        print("error opening file")
-
-def tdee_input():
+            data = json.load(f)
+
+    except FileNotFoundError:
+        with open("data.json", "x") as f:
```

Comment:
Writing data to disk is something we'll want to think about it. Where should it go? 

We don't want to write this JSON file inside the Python package, because this only works in a development install setup. Eventually, we'll want this python package to be distributable, and saving user data to the Python's `site-packages` directly (where installed packages go) isn't a good idea. If they ever uninstall the package their data will get clobbered. 

Often, packages like to install this under a user's home directory in a hidden folder (i.e. folders prefixed with `.`).

Reply:
Good point, didn't really think that one through at all. Or, rather, just made something work and moved on. There's me and my scripting background coming through pretty strongly :)

I'm familiar with that behavior, I can put it in a folder like `.tdee` in the user's home folder or something along those lines.

A little off topic—I'm not sure that a JSON file is the most ideal choice, either. It was easy, and I figure that the amount of data we'll be writing/reading is negligible, but if you think that a sqlite database would be better I could look into that.

Reply:
Okay, made those changes. Let me know what you think!